### PR TITLE
MazeGUI

### PR DESCRIPTION
**Pull request by paige-kohn**

### CODE REVIEW

File: src/main/java/kohn/maze/gui/Cell.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package kohn.maze.gui;
+
+import java.util.ArrayList;
+
+public class Cell {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/gui/Cell.java]: The point of this is not to copy and paste the command line code, it's to be able to use that code in writing your GUI

File: src/main/java/kohn/maze/gui/Maze.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
+package kohn.maze.gui;
+
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Random;
+import java.util.Stack;
+
+public class Maze extends JComponent {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/gui/Maze.java]: You should be creating a MazeView that has a Maze

Comment (paige-kohn) [reply src/main/java/kohn/maze/gui/Maze.java]: Oh I did that, but it doesn't display, so I committed a version that did. II just pushed the one with mazeView.

Comment (paige-kohn) [reply src/main/java/kohn/maze/gui/Maze.java]: Nevermind, it works now.

File: src/main/java/kohn/maze/Maze.java [hunk 3]
```diff
@@ -128,13 +125,38 @@ private boolean containsUnvisitedCells() {
...
-    private void displayMaze() {
+    public int getRows() {
+        return rows;
+    }
+
+    public int getCols() {
+        return cols;
+    }
+
+    public boolean getNorthCell(int x, int y) {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java]: `getNorthCell()` sounds like it's returning the North Cell itself. You want to name this (and the other methods) something like `hasNorthCell()`

File: src/main/java/kohn/maze/Maze.java [hunk 4]
```diff
@@ -158,15 +180,11 @@ private void displayMaze() {
             sb.append("+---");
         }
         sb.append("+");
-        System.out.println();
-        System.out.println(sb.toString());
+        return (sb.toString());
```
Comment (andrewoid) [inline src/main/java/kohn/maze/Maze.java:180]: ()s aren't necessary here.

File: src/main/java/kohn/maze/gui/MazeView.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import javax.swing.*;
+import java.awt.*;
+
+public class MazeView extends JComponent {
+
+    private Maze maze;
+    private int rows;
+    private int cols;
+
+    public void setMaze() {
```
Comment (andrewoid) [inline src/main/java/kohn/maze/gui/MazeView.java]: In general, setters should pass a value.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**